app/rtc_s1_parallel.py

```diff
@@ -0,0 +1,159 @@
+        Path to the original runconfig
+    path_log_in: str
+        Path to the original logfile
+
+    Returns:
+    list_runconfig_burst: list(str)
+        List of the burst runconfigs
+    list_logfile_burst: list(str)
+        List of the burst logfiles,
+        which corresponds to `list_runconfig_burst`
+    '''
+    with open(path_runconfig_in, 'r+', encoding='utf8') as fin:
+        runconfig_dict_in = yaml.safe_load(fin.read())
+
+    cfg = RunConfig.load_from_yaml(path_runconfig_in, 'rtc_s1')
+
+
+    list_runconfig_burst = []
+    list_logfile_burst = []
+
+    # determine the bursrs to process
+    list_burst_id = cfg.bursts.keys()
+
+    for burst_id in list_burst_id:
+        path_temp_runconfig = os.path.join(cfg.scratch_path,
+                                           f'burst_runconfig_{burst_id}.yaml')
+
+        runconfig_dict_out = runconfig_dict_in.copy()
+        runconfig_dict_out['runconfig']['groups']['input_file_group']['burst_id'] = [burst_id]
+        list_runconfig_burst.append(path_temp_runconfig)
+
+        if path_log_in is None:
+            list_logfile_burst.append(None)
+        else:
+            list_logfile_burst.append(f'{path_log_in}.{burst_id}')
+
+        with open(path_temp_runconfig, 'w+', encoding='utf8') as fout:
+            yaml.dump(runconfig_dict_out, fout)
+
+    return list_runconfig_burst, list_logfile_burst
+
+
+def process_runconfig(path_runconfig_burst, path_logfile_burst, full_log_format):
+    '''
+    single worker to process runconfig from terminal using `subprocess`
+
+    Parameters:
+    path_runconfig_burst: str
+        Path to the burst runconfig
+    path_logfile_burst: str
+        Path to the burst logfile
+    full_log_format: bool
+        Enable full formatting of log messages.
+        See `get_rtc_s1_parser()`
+
+    '''
+
+    list_arg_subprocess = ['rtc_s1.py', path_runconfig_burst]
+    if path_logfile_burst is not None: 
+        list_arg_subprocess += ['--log-file', path_logfile_burst]
+
+    if full_log_format:
+        list_arg_subprocess.append('--full-log-format')
+
+    rtnval = subprocess.run(list_arg_subprocess)
+    os.remove(path_runconfig_burst)
+
+    # TODO add log file into the argument for `subprocess.run()`
+
+
+def process_frame_parallel(arg_in):
+    '''
+    Take in the parsed arguments from CLI,
+    split the original runconfign into bursts,
+    and process them concurrently
+
+    Parameter:
+    arg_in: Namespace
+        Parsed argument. See `get_rtc_s1_parser()`
+    '''
+    t0 = time.time()
+    list_burst_runconfig, list_burst_log = split_runconfig(
+                                                arg_in.run_config_path,
+                                                arg_in.log_file)
+
+    with multiprocessing.Pool(arg_in.num_workers) as p:
+        p.starmap(process_runconfig,
+              zip(list_burst_runconfig,
+                  list_burst_log,
+                  [arg_in.full_log_formatting]*len(list_burst_log)))
```

Comment:
You mean something like this?
```suggestion
    repeat_full_log_formatting = repeat(arg_in.full_log_formatting, len(list_burst_log))
    with multiprocessing.Pool(arg_in.num_workers) as p:
        p.starmap(process_runconfig,
              zip(list_burst_runconfig,
                  list_burst_log,
                  repeat_full_log_formatting))
```

Reply:
Yes, it was ok to use `[] * len()`, it's just that the way it was written it was difficult to read. Separating it in a new line helps. Thank you, @seongsujeong

Reply:
Thanks @gshiroma. I prefer to import as less modules as possible, so I will go back to using the list rather than using `itertools`.